src/common/AdaptiveStream.cpp

```diff
@@ -61,36 +61,48 @@ void AdaptiveStream::ResetSegment()
     absolute_position_ = current_rep_->current_segment_->range_begin_;
 }
 
-bool AdaptiveStream::download_segment()
+unsigned int AdaptiveStream::download_segment()
 {
   if (download_url_.empty())
-    return false;
+    return 400;
 
   return download(download_url_.c_str(), download_headers_);
 }
 
+bool AdaptiveStream::retry(unsigned int returnCode)
+{
+  if (returnCode == 200)
+    return false;
+
+  //Don't retry subtitle segment downloads on HTTP 415 Unsupported Media Type
+  if (type_ == AdaptiveTree::SUBTITLE && returnCode == 415)
```

Comment:
You can now add 404 here as well. I got 404 errors on VRT Eén during normal playback and the retries cause repeated errors in logs here as well. So no retries are useful on 404 errors either.

Reply:
what about other 4xx? 5xx?
Why not just return if it's a subtitle (ie - don't retry)
And then fix up the check for 200.
There are more codes that could be an ok response (206 for example)

Reply:
@matthuisman Fine for me as well, although if we want to do this right I would only make 404, 408, 410 and 415 retry the next segment. Everything else should stop handling subtitles altogether.

Reply:
looking at the code - the retry stuff is in it's own thread.
We juts need to tell IA to not wait for retry threads for subs?
Would that be a better fix? So it still retries - but doesn't wait for it.

Reply:
maybe something like this:
https://github.com/peak3d/inputstream.adaptive/pull/445/files
(ill let it build and give it a test)

Reply:
No, I think there is no use in retrying for the same subtitle segment ever if they failed.
Only for 404, 408, 410 and 415 trying the next segment makes sense.
But the easiest definitely is to always try the next segment on failure.

Only THEOplayer is doing this correctly IMO, Shaka player is doing one retry, then
https://github.com/peak3d/inputstream.adaptive/issues/433#issuecomment-626994942